libs/template/config.go

```diff
@@ -171,39 +171,50 @@ func (c *config) promptForValues(r *renderer) error {
 			}
 		}
 
+		// Compute description for the prompt
 		description, err := r.executeTemplate(property.Description)
 		if err != nil {
 			return err
 		}
 
-		// Get user input by running the prompt
-		var userInput string
-		if property.Enum != nil {
-			// convert list of enums to string slice
-			enums, err := property.EnumStringSlice()
-			if err != nil {
-				return err
+		// We wrap this function in a retry loop to allow retries when the user
+		// entered value is invalid.
+		for {
```

Comment:
can you extract this code into separate method for better readability as `promptForValues` became a bit too big to follow

Reply:
Done :)